app/Platform/Controllers/PlayerCompletionProgressController.php

```diff
@@ -0,0 +1,319 @@
+            $userGamesList,
+            $userSiteAwards
+        );
+
+        // If the user is only wanting to see results for a specific console,
+        // filter out every game and award that doesn't have that console ID.
+        if ($targetSystemId) {
+            [$userGamesList, $userSiteAwards] = $this->playerProgressionService->useSystemId(
+                $targetSystemId,
+                $userGamesList,
+                $userSiteAwards
+            );
+        }
+
+        // Remove invalid consoles and add some awards metadata to each entity.
+        $filteredAndJoinedGamesList = $this->playerProgressionService->filterAndJoinGames(
+            $userGamesList,
+            $userSiteAwards,
+        );
+        $primaryCountsMetrics = $this->playerProgressionService->buildPrimaryCountsMetrics($filteredAndJoinedGamesList);
+
+        // The user may only be wanting to see a certain subset of games, such
+        // as only mastered games, or only unawarded games. Perform this filtering now.
+        if ($targetGameStatus) {
+            $filteredAndJoinedGamesList = $this->useGameStatusFilter(
+                $targetGameStatus,
+                $filteredAndJoinedGamesList
+            );
+        }
+
+        $milestones = $this->buildMilestones($filteredAndJoinedGamesList);
+
+        $filteredAndJoinedGamesList = $this->useSortedList($filteredAndJoinedGamesList, $sortOrder);
+
+        $totalInList = count($filteredAndJoinedGamesList);
+        $totalPages = ceil($totalInList / $this->pageSize);
+
+        // Ensure the user isn't trying to go out of bounds.
+        if ($currentPage < 1 || $currentPage > $totalPages) {
+            $currentPage = 1;
+        }
+
+        $paginatedGamesList = array_slice(
+            $filteredAndJoinedGamesList,
+            ($currentPage - 1) * $this->pageSize,
+            $this->pageSize,
+        );
+
+        return view('platform.completion-progress-page', [
+            'allAvailableConsoleIds' => $allAvailableConsoleIds,
+            'completedGamesList' => $paginatedGamesList,
+            'currentPage' => $currentPage,
+            'isFiltering' => $targetGameStatus || $targetSystemId,
+            'me' => $me,
+            'milestones' => $milestones,
+            'primaryCountsMetrics' => $primaryCountsMetrics,
+            'selectedConsoleId' => $targetSystemId,
+            'selectedSortOrder' => $sortOrder,
+            'selectedStatus' => $targetGameStatus,
+            'siteAwards' => $userSiteAwards,
+            'totalInList' => $totalInList,
+            'totalPages' => $totalPages,
+            'user' => $foundTargetUser,
+        ]);
+    }
+
+    private function useSortedList(array $filteredAndJoinedGamesList, string $sortOrder): array
+    {
+        if ($sortOrder === 'unlock_date') {
+            usort($filteredAndJoinedGamesList, function ($a, $b) {
+                return strtotime($b['MostRecentWonDate']) - strtotime($a['MostRecentWonDate']);
```

Comment:
I am running into an error where this array key `MostRecentWonDate` (and sometimes `FirstWonDate`) are undefined for me after testing beating a game in softcore locally and having it be my last completed game

Reply:
Definitely seemed to be cache related after our testing but I wasn't able to reproduce it following the same steps (reset game progress then re-complete it) so hopefully your latest commit will handle that better.